platform-sdk/swirlds-common/src/main/java/com/swirlds/common/crypto/engine/EcdsaSecp256k1Verifier.java

```diff
@@ -165,14 +86,33 @@ public static byte[] minifiedPositiveBigEndian(final byte[] v) {
      * 		the public key to use to verify the signature
      * @return true if the signature is valid
      */
-    public boolean verify(final byte[] rawSig, final byte[] msg, final byte[] pubKey) {
+    public boolean verify(@NonNull final byte[] rawSig, @NonNull final byte[] msg, @NonNull final byte[] pubKey) {
+        // check message is already Keccak256 hash size
+        if (msg.length != ECDSA_KECCAK_256_SIZE) {
+            logger.warn(TESTING_EXCEPTIONS.getMarker(), () -> "Message is not Keccak256 hash size 32 bytes [ msg = %s ]"
+                    .formatted(hex(msg)));
+            return false;
```

Comment:
This new code adds more invalid scenarios that i think should be noted in the javadoc.

Reply:
Nathan and I added more docs